src/main.ts

```diff
@@ -83,6 +83,13 @@ export const LunchMoneyKrakenConnection: LunchMoneyCryptoConnection<
           asset: krakenToCommon(cleaned ?? key),
           amount: value,
         };
+      } else if (balance.type != type) {
+        // Cash types generally start with a Z but the Kraken API returns different symbols
+        // for staking and bonus programs like EUR.M or EUR.HOLD
+        // if we get a Z code for anything let's set the type to cash
+        if (key.startsWith('Z')) {
```

Comment:
I think this would never hit because line 66 has an if statement on `key.startsWith('Z')` already?

Reply:
A single balance entry can include the sum of several balances that all refer to the same currency.  For example, a user could have EUR.M, EUR.HOLD, and ZEUR.   ZEUR is Euros.  The other balances with the dot extensions refer to various bonus or staking programs. Kraken customers have told me that the Kraken UI "sums" these when providing balances, and in fact, that is the original logic in the connector.

This new code addresses a bug in the connector. If EUR.M came first, it would NOT be assigned the type "cash" because it does not start with Z.   By the time the ZEUR symbol is processed, a balancing entry already exists, which has the type "cash."  This new logic always sets the type to "cash" if ANY of the multiple balances that make up the sum starts with Z.

Reply:
This now uses the type returned by the transform logic, as discussed.